supervisor/dbus/udisks2/block.py

```diff
@@ -0,0 +1,262 @@
+        }
+        if not data["fsname"]:
+            data.pop("fsname")
+        return data
+
+
+CrypttabConfigDetailsDataType = TypedDict(
+    "CrypttabConfigurationDetailsDataType",
+    {
+        "name": Required[bytearray],
+        "device": Required[bytearray],
+        "passphrase-path": Required[bytearray],
+        "passphrase-contents": Required[bytearray],
+        "options": Required[bytearray],
+    },
+)
+
+
+@dataclass
+class CrypttabConfigDetails:
+    """crypttab configuration details."""
+
+    device: str
+    passphrase_contents: str
+    options: str
+    name: str | None = None
+    passphrase_path: str | None = None
+
+    @staticmethod
+    def from_dict(data: CrypttabConfigDetailsDataType) -> "CrypttabConfigDetails":
+        """Create CrypttabConfigDetails from dict."""
+        return CrypttabConfigDetails(
+            name=bytes(data["name"]).decode(),
+            device=bytes(data["device"]).decode(),
+            passphrase_path=bytes(data["passphrase-path"]).decode(),
+            passphrase_contents=bytes(data["passphrase-contents"]).decode(),
+            options=bytes(data["options"]).decode(),
+        )
+
+    def to_dict(self) -> dict[str, Variant]:
+        """Return dict representation."""
+        data = {
+            "name": Variant("ay", bytearray(self.name)) if self.name else None,
+            "device": Variant("ay", bytearray(self.device)),
+            "passphrase-path": Variant("ay", bytearray(self.passphrase_path))
+            if self.passphrase_path
+            else None,
+            "passphrase-contents": Variant("ay", bytearray(self.passphrase_contents)),
+            "options": Variant("ay", bytearray(self.options)),
+        }
+        for key in ("name", "passphrase-path"):
+            if not data[key]:
+                data.pop(key)
+        return data
+
+
+class UDisks2Block(DBusInterfaceProxy):
+    """Handle D-Bus interface for UDisks2 block device object.
+
+    http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Block.html
+    """
+
+    name: str = DBUS_IFACE_BLOCK
+    bus_name: str = DBUS_NAME_UDISKS2
+    properties_interface: str = DBUS_IFACE_BLOCK
+
+    def __init__(self, object_path: str) -> None:
+        """Initialize object."""
+        self.object_path = object_path
+        self._additional_interfaces: dict[str, DBusInterfaceProxy] = {}
+        super().__init__()
+
+    async def connect(self, bus: MessageBus) -> None:
+        """Connect to bus."""
+        await super().connect(bus)
+        for key in list(set(self.dbus.proxies) - set(ADDITIONAL_INTERFACES)):
+            iface = self.additional_interfaces[key] = ADDITIONAL_INTERFACES[key](
+                self.object_path
+            )
+            await iface.connect(bus)
+
+    @property
+    def additional_interfaces(self) -> dict[str, DBusInterfaceProxy]:
+        """Return additional interfaces."""
+        return self._additional_interfaces
```

Comment:
I get what you're doing here but it might be easier to understand this interface if we make properties for the specific interfaces. Like a `filesystem` property that returns the filesystem `DBusInterfaceProxy`. I'm fine with the `_additional_interfaces` setup internally just seems like this class would be easier to understand with specific typed properties rather then a dictionary that requires the caller know the exact dbus interface.